src/hera.cpp

```diff
@@ -394,18 +302,41 @@ evmc_result hera_execute(
       ensureCondition(_code.size() > 5, ContractValidationFailure, "Invalid contract or metering failed.");
     }
 
-    execute(context, _code, state_code, *msg, result, meterInterfaceGas);
+    /*
+     * Instantiate, execute, and extract results from the correct WASM engine
+     */
+    ExecutionResult vmresult;
+
+    switch (hera->vm) {
+    #if WABT_SUPPORTED
+    case VM_WABT:
+      WabtVM vm = WabtVM(_code, _state_code, *msg, context, meterInterfaceGas);
+      vm.execute();
+      vmresult = vm.getResult();
+    #endif
+    #if WAVM_SUPPORTED
+    case VM_WAVM:
+      WavmVM vm = WavmVM(_code, _state_code, *msg, context, meterInterfaceGas);
+      vm.execute();
+      vmresult = vm.getResult();
+    #endif
+    default:
```

Comment:
I don't think this should have a default case (or at least the default case should have an assert). Rather only specific ones to the VMs.

Reply:
I will fix that